File: artifacts/flutter-app/lib/models/verse.dart
class Verse {
  final int id;
  final String book;
  final int chapter;
  final int verseNumber;
  final String text;
  final String version;

  const Verse({
    required this.id,
    required this.book,
    required this.chapter,
    required this.verseNumber,
    required this.text,
    required this.version,
  });

  factory Verse.fromJson(Map<String, dynamic> json) {
    return Verse(
      id: json['id'] as int,
      book: json['book'] as String,
      chapter: json['chapter'] as int,
      verseNumber: json['verseNumber'] as int,
      text: json['text'] as String,
      version: json['version'] as String? ?? 'KJV',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'book': book,
      'chapter': chapter,
      'verseNumber': verseNumber,
      'text': text,
      'version': version,
    };
  }
}

class DailyVerseResponse {
  final Verse? verse;
  final String? reflection;

  const DailyVerseResponse({this.verse, this.reflection});

  factory DailyVerseResponse.fromJson(Map<String, dynamic> json) {
    return DailyVerseResponse(
      verse: json['verse'] != null ? Verse.fromJson(json['verse']) : null,
      reflection: json['reflection'] as String?,
    );
  }
}

class VersesResponse {
  final List<Verse> verses;
  final int total;

  const VersesResponse({required this.verses, required this.total});

  factory VersesResponse.fromJson(Map<String, dynamic> json) {
    return VersesResponse(
      verses: (json['verses'] as List?)
              ?.map((v) => Verse.fromJson(v))
              .toList() ??
          [],
      total: json['total'] as int? ?? 0,
    );
  }
}

File: VerseDaily-iOS/Features/Devotional/Domain/Entities/Verse.swift
import Foundation

public struct Verse: Identifiable, Codable, Sendable, Equatable {
    public let id: Int
    public let book: String
    public let chapter: Int
    public let verseNumber: Int
    public let text: String
    public let version: String
    
    public init(id: Int, book: String, chapter: Int, verseNumber: Int, text: String, version: String) {
        self.id = id
        self.book = book
        self.chapter = chapter
        self.verseNumber = verseNumber
        self.text = text
        self.version = version
    }
    
    public var reference: String {
        "\(book) \(chapter):\(verseNumber)"
    }
}

public struct DailyVerse: Sendable, Equatable {
    public let verse: Verse
    public let reflection: String?
    public let date: Date
    
    public init(verse: Verse, reflection: String?, date: Date = Date()) {
        self.verse = verse
        self.reflection = reflection
        self.date = date
    }
}
